src/filesys.cpp

```diff
@@ -740,5 +744,23 @@ bool Rename(const std::string &from, const std::string &to)
 	return rename(from.c_str(), to.c_str()) == 0;
 }
 
+bool bad_char(const char c)
+{
+	return !(isalnum(c) || c == '-' || c == '_' || c == ' ');
```

Comment:
it's simpler that way, previously I got complains about the function being too complex, when I had a blacklist of characters that caused issues